File: app/controllers/walkthrough_controller.rb
class WalkthroughController < UIViewController
  extend IB

  attr_accessor :backgroundImage

  outlet :backgroundImageView, UIImageView
  outlet :dismissButton, UIButton

  def viewDidLoad
    super

    @backgroundImageView.image = @backgroundImage
  end

  def preferredStatusBarStyle
    UIStatusBarStyleLightContent
  end

  def supportedInterfaceOrientations
    UIInterfaceOrientationMaskPortrait
  end

  def skipButton
    @dismissButton.setTitle("Skip", forState: UIControlStateNormal)
  end

  def doneButton
    @dismissButton.setTitle("Done", forState: UIControlStateNormal)
  end

  def dismiss
    self.presentingViewController.dismissModalViewControllerAnimated(true)
  end
end

File: Forking Pasta/WalkthroughController.swift
import UIKit

class WalkthroughController: UIViewController {
    
    var backgroundImage: UIImage?
    
    @IBOutlet weak var backgroundImageView: UIImageView!
    @IBOutlet weak var skipButton: UIButton!
    @IBOutlet weak var doneButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        backgroundImageView.image = backgroundImage
    }
    
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return .LightContent
    }
    
    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask {
        return .Portrait
    }
    
    func showSkipButton() {
        doneButton.hidden = true
        skipButton.hidden = false
    }
    
    func showDoneButton() {
        skipButton.hidden = true
        doneButton.hidden = false
    }
    
    func dismiss() {
        presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
    }
}